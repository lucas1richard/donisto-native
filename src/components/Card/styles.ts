import theme from 'theme/iftheme';
import { StyleSheet } from 'react-native';

/**
 * Function which creates an object to be inserted into StyleSheet
 */

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#fff',
    marginTop: theme.fs,
    padding: theme.fs,
    elevation: 3,
    shadowColor: '#555',
    shadowOpacity: 0.15,
    shadowOffset: {
      width: 2,
      height: 4
    },
    shadowRadius: 6
  }
});

export default styles;
