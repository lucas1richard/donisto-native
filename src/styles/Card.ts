import theme from 'theme/iftheme';

/**
 * Function which creates an object to be inserted into StyleSheet
 */
const card = (specific = {}, colorSpec?: string) => {
  let backgroundColor = theme.white;
  if (colorSpec) {
    backgroundColor = theme[colorSpec];
  }
  if (!backgroundColor) {
    backgroundColor = colorSpec;
  }
  return {
    backgroundColor,
    marginTop: 16,
    padding: 16,
    elevation: 3,
    shadowColor: '#555',
    shadowOpacity: 0.15,
    shadowOffset: {
      width: 2,
      height: 4
    },
    shadowRadius: 6,
    ...specific
  };
};

export default card;
