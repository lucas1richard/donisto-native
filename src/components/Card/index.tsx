import * as React from 'react';
import { View } from 'react-native';
import styles from './styles';

export interface ICardProps {
  children: any;
}

function Card({ children }: ICardProps) {
  return (
    <View style={styles.wrapper}>
      {children}
    </View>
  );
}

export default Card;
