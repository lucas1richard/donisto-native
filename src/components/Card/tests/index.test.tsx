import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card, { ICardProps } from '..';

describe('<Card />', () => {
  const renderComponent = (props: ICardProps) => toJson(shallow(<Card {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({ children: 'mockText' })).toMatchSnapshot();
  });
});