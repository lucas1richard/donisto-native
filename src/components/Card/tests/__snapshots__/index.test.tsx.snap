// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders with just text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    mockText
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "mockText",
      "style": Object {
        "backgroundColor": "#fff",
        "elevation": 3,
        "marginTop": 20,
        "padding": 20,
        "shadowColor": "#555",
        "shadowOffset": Object {
          "height": 4,
          "width": 2,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 6,
      },
    },
    "ref": null,
    "rendered": "mockText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "mockText",
        "style": Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "marginTop": 20,
          "padding": 20,
          "shadowColor": "#555",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 6,
        },
      },
      "ref": null,
      "rendered": "mockText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
